src/lib/utils.ts

```diff
@@ -22,6 +22,23 @@ export const appUrl = readable<string>(PUBLIC_APP_URL, (set) => {
   });
 });
 
+/** Used to ensure that any link to an upgrade includes the partner ID if it is set */
+export const upgradeUrl = readable<string>('/account?action=upgrade', (set) => {
+  partnerData.subscribe((partnerData) => {
+    if (partnerData?.id) {
+      set(`/account?action=upgrade&partnerId=${partnerData.id}`);
+    } else {
+      set('/account?action=upgrade');
+    }
+  });
+});
```

Comment:
I changed some upgrade/checkout links to use this but I did not get all of the because some are already doing some stuff with different query params. Will need to take a deeper look, but I do believe most people will just checkout with the links I changed.